lib/StackAnalysis/StackAnalysis.cpp

```diff
@@ -383,6 +626,818 @@ void commitToModel(GeneratedCodeBasicInfo &GCBI,
+      revng_assert(AbortCall != nullptr
+                   && AbortCall->getCalledFunction() == M.getFunction("abort"));
+      Builder.SetInsertPoint(AbortCall);
+    }
+
+    // Mark end of basic block with a pre-hook call.
+    StructType *MetaAddressTy = MetaAddress::getStruct(&M);
+    SmallVector<Value *, 2> Args = { CallSiteJT.toConstant(MetaAddressTy),
+                                     Next.toConstant(MetaAddressTy) };
+    auto *Last = Builder.CreateCall(PreHookMarker, Args);
+
+    // Prevent the store instructions from being optimized out by storing
+    // the rval of a call to an opaque function into the clobbered registers.
+    RegistersClobberedPool.addFnAttribute(Attribute::ReadOnly);
+    RegistersClobberedPool.addFnAttribute(Attribute::NoUnwind);
+
+    for (GlobalVariable *Reg : ClobberedRegs) {
+      auto *CSVTy = Reg->getType()->getPointerElementType();
+      auto *RegsClobbOpqF = RegistersClobberedPool
+                              .get(BB->getParent()->getName(),
+                                   CSVTy,
+                                   {},
+                                   "registers_clobbered");
+      Builder.CreateStore(Builder.CreateCall(RegsClobbOpqF), Reg);
+    }
+
+    // Adjust back the stack pointer.
+    switch (GCBI->arch()) {
+    case llvm::Triple::x86:
+    case llvm::Triple::x86_64: {
+      auto *SP = Builder.CreateLoad(GCBI->spReg());
+
+      // What is the elected final stack offset?
+      const auto &FSO = Oracle.getElectedFSO(Next);
+      Value *Offset = GCBI->arch() == llvm::Triple::x86 ?
+                        Builder.getInt32(*FSO) :
+                        Builder.getInt64(*FSO);
+      auto *Inc = Builder.CreateAdd(SP, Offset);
+      Builder.CreateStore(Inc, GCBI->spReg());
+      break;
+    }
+    default: {
+      break;
+    }
+    }
+
+    // Mark end of basic block with a post-hook call.
+    Builder.CreateCall(PostHookMarker, Args);
+
+    BB->splitBasicBlock(Last->getPrevNode(),
+                        BB->getName() + Twine("__summary"));
+
+    // Erase the `function_call` marker unless fake.
+    Call->eraseFromParent();
+    break;
+  }
+
+  case FunctionTypeValue::Fake: {
+    // Get the fake function by its entry basic block.
+    Function *FakeFunction = Oracle.getFakeFunction(Next);
+
+    // If fake, it must have been already analyzed.
+    revng_assert(FakeFunction != nullptr);
+
+    // If possible, inline the fake function.
+    auto *CI = CallInst::Create(FakeFunction, "", Term);
+    InlineFunctionInfo IFI;
+    bool Status = InlineFunction(*CI, IFI, nullptr, true).isSuccess();
+    revng_log(StackAnalysisLog,
+              "Has callee " << FakeFunction->getName() << "been inlined? "
+                            << Status);
+    break;
+  }
+
+  default:
+    revng_abort();
+  }
+}
+
+template<class FO>
+llvm::Function *
+CFEPAnalyzer<FO>::createDisposableFunction(llvm::BasicBlock *Entry) {
```

Comment:
How about changing this into `detectFunctionBoundaries`? Similarly, `throwDisposableFunction` into something like `throwOutlinedFunction`. I find _disposable_ a bit meaningless, and who needs to know that it is disposable?

Reply:
`detectFunctionBoundaries` makes me thing you're populating the model, which AFAIU you aren't in this function.
What about
* `createDisposableFunction` -> `isolateFunction`
* `throwDisposableFunction` ->  `discardIsolatedFunction`
?

Reply:
Not sure how detecting the boundaries of function (which is actually what is being done in `createDisposableFunction`)  is correlated with populating the model. I think `isolateFunction` makes more sense, though I think it may cause confusion w/ IsolationFunction pass and whatnot. Maybe `outlineFunction` could work, but I still believe `detectFunctionBoundaries` gives more the idea.

Reply:
There's a bit of historical meaning attached to these terms. `detectFunctionBoundaries` was analysis that stored its results in metadata (kind of the model).
"isolateFunction" is a loaded term too, for the reasons you explained.

We usually called that function "outlined function". So we can go for that. Or, if we want to stick to LLVM terminology (`CodeExtractor`) we can go for "extracted function".